File: insertionSort.c
#include<stdio.h>
#include<stdlib.h>

void main(){
    int n;
    printf("\n----------INSERTION SORT----------\n\n");
    printf("Enter array size: ");
    scanf("%d",&n);
    int *array;
    array = (int*)malloc(sizeof(int)*n);
    printf("Insert array elements: ");

    for(int i=0;i<n;i++){
        scanf("%d",&array[i]);
    }

    int i,j,c;
    //insertion sort
    for(i=1;i<n;i++){
        c = array[i];
        j = i-1;
        while(j>=0 && c < array[j]){
            array[j + 1] = array[j];
            j-=1;
        }
        array[j+1] = c;
    }
    
    printf("Sorted array: ");
    for(int i=0;i<n;i++){
        printf("%d ",array[i]);
    }
    printf("\n");
    printf("\n--------------EXIT--------------\n");
}

File: Python/insertionSort.py
#Python program to perform insertion sort
def insertionSort(array):
    n = len(array)
    for i in range(n):
        c = array[i]
        j = i-1
        while(j>=0 and c < array[j]):
            array[j+1] = array[j]
            j = j-1
        array[j+1] = c

if __name__ == "__main__":
    print("\n------INSERTION SORT------\n")
    n = int(input("Enter array size: "))

    array = []

    print("Insert array elements: ")
    for i in range(n):
        ele = int(input())
        array.append(ele)

    insertionSort(array)

    print("Sorted array: ")
    print(array)